packages/babel-types/src/definitions/experimental.js

```diff
@@ -258,3 +258,25 @@ defineType("BigIntLiteral", {
   },
   aliases: ["Expression", "Pureish", "Literal", "Immutable"],
 });
+
+defineType("SliceExpression", {
+  visitor: ["object", "lower", "upper", "step"],
+  aliases: ["Expression", "LVal"],
+  fields: {
+    object: {
+      validate: assertNodeType("Expression"),
+    },
+    lower: {
+      validate: assertNodeType("Identifier", "NumericLiteral"),
```

Comment:
This can't be a generic expression? It can at least be a `UnaryExpression`.

Reply:
> This can't be a generic expression?

Currently the proposal [restricts](https://github.com/tc39/proposal-slice-notation#can-the-upper-bound-lower-bound-or-the-step-argument-be-an-arbitrary-expression) the argument to be IdentifierReference and DecimalDigits.

> It can at least be a UnaryExpression

Yes given that it can be a negative number or negative infinity. I should have added tests on that. 🤦‍♂️